tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py

```diff
@@ -29,146 +34,144 @@ def setUp(self):
+        Code can currently only handle one record per invocation.
         """
-        multipart_chunksize_mb = random.randint(1, 10000)
-        mock_execute = Mock(
-            return_value=[("1", "3", "f1.doc", "s3://restore", multipart_chunksize_mb)]
-        )
-        mock_connection = Mock()
-        mock_connection.execute = mock_execute
-        mock_exit = Mock()
-        mock_enter = Mock()
-        mock_enter.__enter__ = Mock(return_value=mock_connection)
-        mock_enter.__exit__ = mock_exit
-        mock_engine = Mock()
-        mock_engine.begin = Mock(return_value=mock_enter)
-        mock_get_user_connection.return_value = mock_engine
-
-        record = self.event["Records"][0]
-        new_data = {
-            post_copy_request_to_queue.JOB_ID_KEY: "1",
-            post_copy_request_to_queue.GRANULE_ID_KEY: "3",
-            post_copy_request_to_queue.FILENAME_KEY: "f1.doc",
-            # todo: Value incorrect here and elsewhere. As written, must match mock_execute's return value.
-            post_copy_request_to_queue.RESTORE_DESTINATION_KEY: "s3://restore",
-            post_copy_request_to_queue.MULTIPART_CHUNKSIZE_MB_KEY: multipart_chunksize_mb,
-            post_copy_request_to_queue.TARGET_KEY_KEY: "b21b84d653bb07b05b1e6b33684dc11b",
-            post_copy_request_to_queue.SOURCE_KEY_KEY: "b21b84d653bb07b05b1e6b33684dc11b",
-            post_copy_request_to_queue.SOURCE_BUCKET_KEY: "lambda-artifacts-deafc19498e3f2df",
+        event = {
+            "Records": [
+                {
+                    "s3": {
+                        "object": {"key": uuid.uuid4().__str__()},
+                        "bucket": {"name": uuid.uuid4().__str__()},
+                    }
+                },
+                {
+                    "s3": {
+                        "object": {"key": uuid.uuid4().__str__()},
+                        "bucket": {"name": uuid.uuid4().__str__()},
+                    }
+                },
+            ]
         }
-        backoff_args = [
-            self.db_queue_url,
-            self.recovery_queue_url,
-            2,
-            2,
-            2,
-        ]
-        # calling the task function
-        task(record, *backoff_args)
 
-        mock_get_user_connection.assert_called_once_with(
-            mock_get_configuration.return_value
-        )
-        mock_execute.assert_called_once_with(mock_get_metadata_sql.return_value)
-        mock_update_status_for_file.assert_called_once_with(
-            "1",
-            "3",
-            "f1.doc",
-            shared_recovery.OrcaStatus.STAGED,
-            None,
-            self.db_queue_url,
-        )
-        mock_post_entry_to_queue.assert_called_with(
-            new_data,
-            shared_recovery.RequestMethod.NEW_JOB,
-            self.recovery_queue_url,
-        )
-        mock_exit.assert_called_once()
+        with self.assertRaises(ValueError) as cm:
+            with patch.dict(
```

Comment:
I don't like how it moves definition of values that should be randomized into an environment variable. Then you have to manually pull them out of os.environ and into storage variables for checking, and you'll have to cast them as well, since os.environ only holds strings.